### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by RLCorp**
## Checklist

- [ ] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts [hunk 1]
```diff
@@ -35,13 +124,49 @@ describe('TestReportDashboardPage', () => {
...
+        MockComponent(SignatureComponent),
+        MockComponent(ActivityCodeComponent),
+        MockComponent(TransmissionDisplayComponent),
+        MockComponent(CPCDebriefCardComponent),
+        MockComponent(Adi3DebriefCard),
+        MockComponent(Adi3DebriefCardBox),
+        MockComponent(SearchablePicklistComponentWrapper),
+        MockComponent(SearchablePicklistModal),
+        MockComponent(DisplayAddressComponent),
+        MockComponent(DrivingFaultsBadgeComponent),
```
Comment (matthew2564) [inline src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts]: Are ALL of these relevant?

File: src/app/pages/test-report/cat-a-mod1/components/activity-code-4-modal/__tests__/activity-code-4-modal.spec.ts [hunk 2]
```diff
@@ -54,4 +55,19 @@ describe('ActivityCode4Modal', () => {
     });
   });
 
+  describe('onCancel', () => {
+    it('should call dismiss with CANCEL', async () => {
+      spyOn(component['modalController'], 'dismiss');
```
Comment (matthew2564) [inline src/app/pages/test-report/cat-a-mod1/components/activity-code-4-modal/__tests__/activity-code-4-modal.spec.ts]: Can we not use this syntax, mentioned before to access private providers we use this pattern

Declare
`let modalController: ModalController;`

Add this to beforeEach setup

`modalController = TestBed.inject(ModalController);`

Then
`spyOn(modalController, 'dismiss);`

File: src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts [hunk 3]
```diff
@@ -35,13 +124,49 @@ describe('TestReportDashboardPage', () => {
         MockComponent(DashboardItemComponent),
         MockComponent(PracticeModeBanner),
         MockComponent(TimerComponent),
+        MockComponent(DateTimeInputComponent),
```
Comment (andrewsetterfield) [inline src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts]: this is a big list. are they all needed to satify your new tests?

File: src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts]: addressed in newest commit

File: src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts (hunk 3 above)
Comment (RLCorp) [reply src/app/pages/test-report-dashboard/__tests__/test-report-dashboard.spec.ts]: addressed in newest commit

File: src/app/pages/test-report/cat-a-mod1/components/activity-code-4-modal/__tests__/activity-code-4-modal.spec.ts (hunk 2 above)
Comment (RLCorp) [reply src/app/pages/test-report/cat-a-mod1/components/activity-code-4-modal/__tests__/activity-code-4-modal.spec.ts]: they have been replaced with the suggested method

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**